evap/staff/tests/test_views.py

```diff
@@ -215,6 +215,26 @@ def test_reward_points_granting_message(self):
         )
 
 
+class TestUserDeleteView(WebTestStaffMode):
```

Comment:
TestUserDeleteView, TestCourseDeleteView, TestEvaluationDeleteView share the exact same structure. It might make sense to and deduplicate their logic (into a base class?)

That could even be extended to TestSemesterArchiveParticipationsView or GradeDeleteTest, although they are slightly more different.

I'll play around and see if I can find something promising.

Reply:
@niklasmohrin (and anyone else who cares): I put a mockup here: https://github.com/He3lixxx/EvaP/commit/4413db8b09fbe0e2c0029c08b22f2d54e0bc4f94. I think it works quite well for TestUserDeleteView, TestSemesterDeleteGradeDocumentsView, TestCourseDeleteView, and TestQuestionnaireDeleteView, and it works somewhat okay for TestSemesterDeleteView (where we have one additional test I wouldn't want to leave out). What do you think?

It would be nice if we could do the same thing with these tests, but that would be more complicated:
* EvaluationDelete - has significant other assertions that I'd like to keep
* ArchiveParticipations -- doesn't delete a db object, but asserts relevant model method is called
* GradeDelete - no "bad request" case, always succeeds

The test runner will find all classes that inherit from unittest.TestCase, so we can't make the Mixin derive from WebTest -- that's why I chose the current mixin design.

Reply:
I like :+1:

Reply:
Added the commit.